src/kubernetes.rs

```diff
@@ -209,4 +216,105 @@ impl<'a> Kubernetes<'a> {
     pub fn create_selector(&self, key: &str, value: &str) -> BTreeMap<String, String> {
         k8s_helpers::create_selector(key, value)
     }
+
+    pub async fn deploy_replicas_set(
+        &self,
+        replica_set: &ReplicaSet,
+    ) -> Result<ReplicaSet, kube::Error> {
+        let api: Api<ReplicaSet> =
+            Api::namespaced(self.k8s_client.clone(), self.namespace.as_str());
+        let post_params = PostParams::default();
+        // Apply the ReplicaSet
+        api.create(&post_params, replica_set).await
+    }
+
+    pub fn create_bootstrap_service(
+        &self,
+        service_name: &str,
+        label_selector: &BTreeMap<String, String>,
+    ) -> Service {
+        k8s_helpers::create_service(
+            service_name.to_string(),
+            self.namespace.clone(),
+            label_selector.clone(),
+            false,
+        )
+    }
+
+    pub async fn deploy_service(&self, service: &Service) -> Result<Service, kube::Error> {
+        let post_params = PostParams::default();
+        // Create an API instance for Services in the specified namespace
+        let service_api: Api<Service> =
+            Api::namespaced(self.k8s_client.clone(), self.namespace.as_str());
+
+        // Create the Service object in the cluster
+        service_api.create(&post_params, service).await
+    }
+
+    pub fn create_validator_load_balancer(
+        &self,
+        service_name: &str,
+        label_selector: &BTreeMap<String, String>,
+    ) -> Service {
+        k8s_helpers::create_service(
+            service_name.to_string(),
+            self.namespace.clone(),
+            label_selector.clone(),
+            true,
+        )
+    }
+
+    pub async fn check_replica_set_ready(
+        &self,
+        replica_set_name: &str,
+    ) -> Result<bool, kube::Error> {
```

Comment:
micro-nit: the word `check` doesn't make it clear what this returns, how about changing this to `is_replica_set_ready`? Otherwise, you can have this return an enum with `Ready` and `NotReady` as the states, to force people to match on it